File: RemoveElement.java

public class RemoveElement {
	public static int removeElement(int[] nums, int val) {
		int cur = 0;
		for (int i = 0; i < nums.length; i++) {
			if (nums[i] != val) {
				nums[cur++] = nums[i];
			}
		}
	    return cur;
    }
	public static void main(String[] args) {
		int[] a = {1,3,5,1,4};
		System.out.println(removeElement(a, 1));
	}
}

File: py_solution/remove_element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        top = -1
        for i in range(len(nums)):
            if nums[i] != val:
                top += 1
                nums[top] = nums[i]
        return top + 1


if __name__ == "__main__":
    nums = [3, 2, 2, 3]
    solution = Solution()
    print(solution.removeElement(nums, 3), nums)